### ISSUE TITLE
Adding a blog that already exists

### ISSUE THREAD

**Issue by amigre38** [johanneszab/TumblThree#29]
I encounter this one in the past but it's not corrected in the 20 latest version.
If you add in your list a blog that exists and that you already download. TumblThree tell you the blog exists in the list.

The next time you relaunch the program the blog appears as if it was never downloaded (no last download time and no counter of files downloaded) so you need to download all the blog another time.

**Comment by johanneszab** [comment]
Yap, you've written me an email about this. I couldn't reproduce it, so I cannot fix it. Anything special you do? Whats your Windows version? Where do you store your blogs? In some special location?

For me, If I add a blog which is already in the list, I get the notification on top, and that's it. The blog already in the list isn't affected regarding its actual state (active crawl or not). After the next startup, the values are the same as they were before and are not zeroed.

I also couldn't reproduce #9. Maybe it's somehow related. I've tested NTFS and Fat32 formatted drives, no differences.

**Comment by amigre38** [comment]
I see where the problem is in the code:
It's in the procedure SaveBlog , must be in the procedure SaveTumblrFiles too.

After this instruction:
                    File.Replace(newIndex, currentIndex, backupIndex, true);

The replace here is done and then the .bak (containing the good tumblr file with the item downloaded by a new one where the item downloaded is set to 0) is deleted. It's strange you don't have it, i try in the project with visual studio and can reproduce each time.

I think here the newIndex mustn't be created as that is not the same as when a download occurs and you want to save the progress.

**Comment by amigre38** [comment]
My windows version is Windows 10, the blog is on an external drive but mounted with a unit, it's in NTFS else nothing special. I can reproduce on 2 different drive. So i think it's more or less related to the code than the hardware.

**Comment by johanneszab** [comment]
I see. I've moved the [code down from here](https://github.com/johanneszab/TumblThree/blob/575bb7ab10656cddcb6f5c660a0090985e64cd27/src/TumblThree/TumblThree.Applications/Controllers/ManagerController.cs#L924) in order to make sure the files are saved. Thats probably were the error was introduced

**Comment by johanneszab** [comment]
Yes, the _files.tumblr are not removed, I've also noticed that yesterday, but only after the upload. As I've said earlier, I'm actually working the whole day right now to get things done before a deadline and hardly have time and/or concentration in the evenings for this. I've also not really used the program for a year or so.

If you want to fix the throttling issue, you can code it if you want?

**Comment by amigre38** [comment]
I'm not completely fluent in this langage code, i can't make a promise but if i have time, i can try to work on the things i see. ;)

**Comment by amigre38** [comment]
I update the code in my fork to correct the three little things i have.
I let you review and integrate in the real project if you are ok with it!

Thanks ;)

**Comment by johanneszab** [comment]
Okay, do it. Sound good :)

Most likely I'll accept anything ;).

As for the new api limit tumblr introduced:
I've already thought of doing the api access in batches, like crawling for 300 urls initially, then download those images/videos, and crawl in parallel for the next 300 pictures etc to prevent hitting the api limit. In that case, the time/bandwidth it not wasted but is already downloading the blog.

We could also save the most recent post id if the blog was crawled once completely and for any re-crawls stop at that id once we've step over it. There is most likely no need to re-crawl everything.

Or we just skip the api in total and use the webpage. I've already implemented this for picture downloading using the html agility pack. I can upload this in a different branch so you can look into that if you're interested. We'll probably have to skip the api anyways if we ever want to download private/password protected blogs.

**Comment by amigre38** [comment]
I can have a look, it may be difficult for me but i can learn and try to help you if i can!
I must try to understand how it works at this time with the api first i think.

**Comment by johanneszab** [comment]
No hurry!

TumblThree is back to a functional state right now. A bit slower, but it's working. That's why I kinda rushed with those latest unfinished releases. 

And for the api change, I know it's quite a big task. You don't have to do this. I'm happy for everyone who wants to commit anything. Do what you like. You might have other priorities or things you'd like to see improved or you like to code. You could also fix #21 as you've basically already explained it in the comments there, you just have to put it into code.

If you want to change the user interface, do it. For me as someone who knows how everything is working internally it's always hard to come up with an interface which is clear for the user without that knowledge. If you like doing UI stuff, you can change everything there.

Basically, take your time, do what you want. I'm happy for any help. It's always good to have at least 3-4 people coding ..
